assignments/array-methods.js

```diff
@@ -56,28 +56,64 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+let newfullName = runners.forEach(function(i){
+    fullName.push(i.first_name + " " + i.last_name);
+});
 console.log(fullName);
 
+
+
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+let allCaps  = runners.map( function(item){
+    return item.first_name.toUpperCase();
+
+});
+
 console.log(allCaps); 
 
+
+
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter(largeShirts => largeShirts["shirt_size"] === "L");
+    
+
+;
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((theReducer, item) =>{
+    return theReducer += item.donation;}, 0);
+
+
+
 console.log(ticketPriceTotal);
 
+
+
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
 
+// Director sobered up and wanted everyones first name in lower case
+
+let newCaps  = runners.map( function(item){
+    return item.first_name.toLowerCase();
+
+});
+
+console.log(newCaps);
+
 // Problem 2
 
-// Problem 3
\ No newline at end of file
+let smallShirts = runners.filter(largeShirts => largeShirts["shirt_size"] === "M");
+    
```

Comment:
Using the array name small shirts, and then large shirts inside the filter function is a little confusing. Also keep in mind that the filter function loops through every shirt one at a time, so something like `shirt` might be a better variable/descriptor for inside your anonymous function.